src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,555 @@
+//! This module provides a view for breadcrumbs, enabling us to know which node the graph being
+//! edited belongs to and navigating through them.
+
+use crate::prelude::*;
+
+pub mod breadcrumb;
+pub mod project_name;
+
+pub use breadcrumb::Breadcrumb;
+pub use project_name::ProjectName;
+
+use crate::graph_editor::LocalCall;
+
+use enso_frp as frp;
+use enso_protocol::language_server::MethodPointer;
+use enso_protocol::language_server::types::Path;
+use ensogl::data::color;
+use ensogl::display;
+use ensogl::display::camera::Camera2d;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::*;
+use ensogl::display::shape::text::text_field::FocusManager;
+use ensogl::display::Sprite;
+use ensogl::gui::component;
+use logger::AnyLogger;
+use logger::enabled::Logger;
+use std::cmp::Ordering;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+// FIXME[dg] hardcoded literal for glyph of height 12.0. Copied from port.rs
+const GLYPH_WIDTH       : f32 = 7.224_609_4;
+const VERTICAL_MARGIN   : f32 = GLYPH_WIDTH;
+const HORIZONTAL_MARGIN : f32 = GLYPH_WIDTH;
+const TEXT_SIZE         : f32 = 12.0;
+
+
+
+// ========================
+// === RelativePosition ===
+// ========================
+
+#[derive(Debug,Clone,Copy)]
+enum RelativePosition {
+    LEFT,
+    RIGHT
+}
+
+impl Default for RelativePosition {
+    fn default() -> Self {
+        RelativePosition::RIGHT
+    }
+}
```

Comment:
is there any explanation why default relative position is right, not left? Please add it as a short doc comment here.

Reply:
@wdanilo I removed the default initialization for `RelativePosition` and made `relative_position` optional.